<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>morphologyExWithCVClass</class>
 <widget class="QWidget" name="morphologyExWithCVClass">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>400</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>morphologyExWithCV</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="1" column="4">
    <widget class="QPushButton" name="pushButton_5">
     <property name="text">
      <string>闭运算</string>
     </property>
    </widget>
   </item>
   <item row="2" column="2">
    <widget class="QPushButton" name="pushButton_7">
     <property name="text">
      <string>外部梯度</string>
     </property>
    </widget>
   </item>
   <item row="1" column="3">
    <widget class="QPushButton" name="pushButton_4">
     <property name="text">
      <string>开运算</string>
     </property>
    </widget>
   </item>
   <item row="1" column="0">
    <widget class="QPushButton" name="pushButton">
     <property name="text">
      <string>原图</string>
     </property>
    </widget>
   </item>
   <item row="1" column="1">
    <widget class="QPushButton" name="pushButton_2">
     <property name="text">
      <string>膨胀</string>
     </property>
    </widget>
   </item>
   <item row="2" column="4">
    <widget class="QPushButton" name="pushButton_10">
     <property name="text">
      <string>Y向梯度</string>
     </property>
    </widget>
   </item>
   <item row="2" column="1">
    <widget class="QPushButton" name="pushButton_6">
     <property name="text">
      <string>内部梯度</string>
     </property>
    </widget>
   </item>
   <item row="1" column="5">
    <spacer name="horizontalSpacer">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>40</width>
       <height>20</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="4" column="0" colspan="6">
    <widget class="QSlider" name="horizontalSlider">
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>15</number>
     </property>
     <property name="sliderPosition">
      <number>3</number>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="invertedControls">
      <bool>false</bool>
     </property>
     <property name="tickPosition">
      <enum>QSlider::TicksAbove</enum>
     </property>
     <property name="tickInterval">
      <number>1</number>
     </property>
    </widget>
   </item>
   <item row="1" column="2">
    <widget class="QPushButton" name="pushButton_3">
     <property name="text">
      <string>腐蚀</string>
     </property>
    </widget>
   </item>
   <item row="2" column="3">
    <widget class="QPushButton" name="pushButton_9">
     <property name="text">
      <string>X向梯度</string>
     </property>
    </widget>
   </item>
   <item row="0" column="0" colspan="6">
    <widget class="pGraphicsViewer" name="graphicsView"/>
   </item>
   <item row="2" column="0">
    <widget class="QPushButton" name="pushButton_8">
     <property name="text">
      <string>梯度原图</string>
     </property>
    </widget>
   </item>
   <item row="3" column="0">
    <widget class="QPushButton" name="pushButton_11">
     <property name="text">
      <string>hat原图</string>
     </property>
    </widget>
   </item>
   <item row="3" column="1">
    <widget class="QPushButton" name="pushButton_12">
     <property name="text">
      <string>直接二值化</string>
     </property>
    </widget>
   </item>
   <item row="3" column="2">
    <widget class="QPushButton" name="pushButton_13">
     <property name="text">
      <string>黑帽后二值化</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>pGraphicsViewer</class>
   <extends>QGraphicsView</extends>
   <header>../GraphicsViewer/pGraphicsViewer.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="morphologyExWithCV.qrc"/>
 </resources>
 <connections/>
</ui>
